File: WebSite with CosmosDB/server/Basketball/Microservice/Infrastructure/IBasketballRepo.cs
using System.Linq.Expressions;

namespace NBA.Infrastructure
{
    interface IBasketballRepo
    {
        Task<T> CreateAsync<T>(T entity) where T : BasketballModel;
        Task<T> UpdateAsync<T>(T entity) where T : BasketballModel;
        T GetById<T>(string id) where T : BasketballModel;
        IEnumerable<T> Get<T>(Expression<Func<T, bool>> where, Expression<Func<T, object>> order = null, int? take = null) where T : BasketballModel;
        void Reseed();
    }
}

File: Java with MongoDB/server/Basketball/Microservice/basketball-microservice/src/main/java/com/nba/basketball_microservice/infrastructure/IBasketballRepo.java
package com.nba.basketball_microservice.infrastructure;

import java.util.List;
import java.util.function.Function;
import org.bson.conversions.Bson;

public interface IBasketballRepo {

    <T extends BasketballModel> T create(T entity);

    <T extends BasketballModel> T update(T entity);

    <T extends BasketballModel> T getById(String id, Class<T> clazz);

    public <T extends BasketballModel> List<T> get(Class<T> clazz, Bson filter, Function<Class<T>, String> order,
            boolean descending, Integer take);

    void reseed();
}